Do not issue an error message when unable to remove .cache on read-only fs.
===================================================================
RCS file: /cvs/lvm2/LVM2/lib/filters/filter-persistent.c,v
retrieving revision 1.49
retrieving revision 1.50
diff -u -r1.49 -r1.50
--- LVM2/lib/filters/filter-persistent.c	2011/04/22 12:05:33	1.49
+++ LVM2/lib/filters/filter-persistent.c	2011/05/12 12:42:48	1.50
@@ -108,7 +108,7 @@
 			log_very_verbose("Obtaining device list from "
 					 "udev. Removing obolete %s.",
 					 pf->file);
-			if (unlink(pf->file) < 0)
+			if (unlink(pf->file) < 0 && errno != EROFS)
 				log_sys_error("unlink", pf->file);
 		}
 		return 1;
